### PR TITLE
Automate Konflux apply configurations for every repo and branch

### PR DESCRIPTION
**Pull request by pierDipi**
Allow with a single command to apply every application, component, and image repository for every repo and branch

See `Apply Konflux configurations` in `README.md` for details on how it can be used

Tested with:

```shell
$ make konflux-apply
rm -rf openshift openshift-knative
go run github.com/openshift-knative/hack/cmd/konflux-apply
2024/08/06 16:02:00 Mirroring repository openshift-knative/backstage-plugins
Cloning into bare repository 'openshift-knative/backstage-plugins/.git'...
remote: Enumerating objects: 7665, done.
remote: Counting objects: 100% (1009/1009), done.
remote: Compressing objects: 100% (400/400), done.
remote: Total 7665 (delta 752), reused 609 (delta 609), pack-reused 6656
Receiving objects: 100% (7665/7665), 9.51 MiB | 7.96 MiB/s, done.
Resolving deltas: 100% (3136/3136), done.
2024/08/06 16:02:03 Mirroring repository openshift-knative/eventing-hyperfoil-benchmark
Cloning into bare repository 'openshift-knative/eventing-hyperfoil-benchmark/.git'...
remote: Enumerating objects: 16704, done.
remote: Counting objects: 100% (775/775), done.
remote: Compressing objects: 100% (116/116), done.
remote: Total 16704 (delta 679), reused 735 (delta 653), pack-reused 15929
Receiving objects: 100% (16704/16704), 3.37 MiB | 1.30 MiB/s, done.
Resolving deltas: 100% (15878/15878), done.
2024/08/06 16:02:06 Mirroring repository openshift-knative/eventing-istio
Cloning into bare repository 'openshift-knative/eventing-istio/.git'...
remote: Enumerating objects: 18496, done.
remote: Counting objects: 100% (3405/3405), done.
remote: Compressing objects: 100% (1486/1486), done.
remote: Total 18496 (delta 1841), reused 3076 (delta 1699), pack-reused 15091
Receiving objects: 100% (18496/18496), 18.90 MiB | 10.75 MiB/s, done.
Resolving deltas: 100% (9336/9336), done.
Switched to branch 'release-next'
2024/08/06 16:02:10 Mirroring repository openshift-knative/eventing-kafka-broker
Cloning into bare repository 'openshift-knative/eventing-kafka-broker/.git'...
remote: Enumerating objects: 87485, done.
remote: Counting objects: 100% (4936/4936), done.
remote: Compressing objects: 100% (2385/2385), done.
remote: Total 87485 (delta 2262), reused 4349 (delta 1940), pack-reused 82549
Receiving objects: 100% (87485/87485), 49.17 MiB | 14.38 MiB/s, done.
Resolving deltas: 100% (49224/49224), done.
Switched to branch 'release-next'
2024/08/06 16:02:17 Mirroring repository openshift-knative/eventing
Cloning into bare repository 'openshift-knative/eventing/.git'...
remote: Enumerating objects: 129400, done.
remote: Counting objects: 100% (503/503), done.
remote: Compressing objects: 100% (403/403), done.
remote: Total 129400 (delta 256), reused 288 (delta 85), pack-reused 128897
Receiving objects: 100% (129400/129400), 82.57 MiB | 12.64 MiB/s, done.
Resolving deltas: 100% (84584/84584), done.
Switched to branch 'release-next'
imagerepository.appstudio.redhat.com/knative-eventing-apiserver-receive-adapter-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-appender-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-channel-controller-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-channel-dispatcher-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-controller-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-event-display-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-filter-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-heartbeats-receiver-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-heartbeats-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-ingress-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-jobsink-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-migrate-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-mtchannel-broker-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-mtping-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-pong-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-schema-release-next configured
imagerepository.appstudio.redhat.com/knative-eventing-webhook-release-next configured
component.appstudio.redhat.com/knative-eventing-apiserver-receive-adapter-release-next configured
component.appstudio.redhat.com/knative-eventing-appender-release-next configured
component.appstudio.redhat.com/knative-eventing-channel-controller-release-next configured
component.appstudio.redhat.com/knative-eventing-channel-dispatcher-release-next configured
component.appstudio.redhat.com/knative-eventing-controller-release-next configured
component.appstudio.redhat.com/knative-eventing-event-display-release-next configured
component.appstudio.redhat.com/knative-eventing-filter-release-next configured
component.appstudio.redhat.com/knative-eventing-heartbeats-receiver-release-next configured
component.appstudio.redhat.com/knative-eventing-heartbeats-release-next configured
component.appstudio.redhat.com/knative-eventing-ingress-release-next configured
component.appstudio.redhat.com/knative-eventing-jobsink-release-next configured
component.appstudio.redhat.com/knative-eventing-migrate-release-next configured
component.appstudio.redhat.com/knative-eventing-mtchannel-broker-release-next configured
component.appstudio.redhat.com/knative-eventing-mtping-release-next configured
component.appstudio.redhat.com/knative-eventing-pong-release-next configured
component.appstudio.redhat.com/knative-eventing-schema-release-next configured
component.appstudio.redhat.com/knative-eventing-webhook-release-next configured
application.appstudio.redhat.com/serverless-operator-release-next unchanged
2024/08/06 16:02:49 Mirroring repository openshift-knative/serverless-operator
Cloning into bare repository 'openshift-knative/serverless-operator/.git'...
remote: Enumerating objects: 122334, done.
remote: Counting objects: 100% (4249/4249), done.
remote: Compressing objects: 100% (1929/1929), done.
remote: Total 122334 (delta 2144), reused 3936 (delta 1998), pack-reused 118085
Receiving objects: 100% (122334/122334), 82.21 MiB | 14.09 MiB/s, done.
Resolving deltas: 100% (71329/71329), done.
2024/08/06 16:02:58 Mirroring repository openshift-knative/net-istio
Cloning into bare repository 'openshift-knative/net-istio/.git'...
remote: Enumerating objects: 44164, done.
remote: Counting objects: 100% (4791/4791), done.
remote: Compressing objects: 100% (1546/1546), done.
remote: Total 44164 (delta 3267), reused 4349 (delta 3041), pack-reused 39373
Receiving objects: 100% (44164/44164), 37.08 MiB | 12.46 MiB/s, done.
Resolving deltas: 100% (27974/27974), done.
2024/08/06 16:03:04 Mirroring repository openshift-knative/net-kourier
Cloning into bare repository 'openshift-knative/net-kourier/.git'...
remote: Enumerating objects: 51722, done.
remote: Counting objects: 100% (1335/1335), done.
remote: Compressing objects: 100% (650/650), done.
remote: Total 51722 (delta 667), reused 1242 (delta 585), pack-reused 50387
Receiving objects: 100% (51722/51722), 47.96 MiB | 12.42 MiB/s, done.
Resolving deltas: 100% (30080/30080), done.
2024/08/06 16:03:10 Mirroring repository openshift-knative/serving
Cloning into bare repository 'openshift-knative/serving/.git'...
remote: Enumerating objects: 169193, done.
remote: Counting objects: 100% (13564/13564), done.
remote: Compressing objects: 100% (790/790), done.
remote: Total 169193 (delta 13084), reused 12792 (delta 12772), pack-reused 155629
Receiving objects: 100% (169193/169193), 95.84 MiB | 14.86 MiB/s, done.
Resolving deltas: 100% (114261/114261), done.
Switched to branch 'release-next'
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,31 +13,48 @@ CI tooling and hacks to improve CI
...
+
 - It is not currently possible to disable periodics per job
 
+## Apply Konflux configurations
+
+1. Follow the instructions to access the Konflux instance
+   at [GitLab (VPN required)](https://gitlab.cee.redhat.com/konflux/docs/users/-/blob/main/topics/getting-started/getting-access.md#accessing-konflux-via-cli)
+2. Set the `kubectl` context to use the `ocp-serverless` workspace
+    ```shell
+    kubectl config set-context konflux-ocp-serverless
```
Comment (creydr) [inline README.md:54]:
```suggestion
    kubectl config use-context konflux-ocp-serverless
```

**Review by creydr — COMMENTED** [review]
One nit in the readme. Otherwise LGTM (tested with eventing-istio).

Thanks for automating this :+1:

File: README.md (hunk 1 above)
Comment (pierDipi) [reply README.md:54]: I don't see `use-context` in my kubectl

```
Available Commands:
  current-context   Display the current-context
  delete-cluster    Delete the specified cluster from the kubeconfig
  delete-context    Delete the specified context from the kubeconfig
  delete-user       Delete the specified user from the kubeconfig
  get-clusters      Display clusters defined in the kubeconfig
  get-contexts      Describe one or many contexts
  get-users         Display users defined in the kubeconfig
  rename-context    Rename a context from the kubeconfig file
  set               Set an individual value in a kubeconfig file
  set-cluster       Set a cluster entry in kubeconfig
  set-context       Set a context entry in kubeconfig
  set-credentials   Set a user entry in kubeconfig
  unset             Unset an individual value in a kubeconfig file
  use-context       Set the current-context in a kubeconfig file
  view              Display merged kubeconfig settings or a specified kubeconfig file

Usage:
  kubectl config SUBCOMMAND [options]

Use "kubectl config <command> --help" for more information about a given command.
Use "kubectl options" for a list of global command-line options (applies to all commands).

```

Comment (pierDipi) [reply README.md]: Thanks, good catch!

**Review by creydr — APPROVED** [review]
/lgtm

**Pull request merged**